components/MainLayout.js

```diff
@@ -10,55 +11,83 @@ const ga = `
   ga('send', 'pageview');
 `
 
-export default function MainLayout ({ children }) {
-  return (
-    <div>
-      <Head>
-        <title>React Bangkok 2.0.0</title>
-        <meta
-          name='viewport'
-          content='width=device-width, initial-scale=1.0'
-        />
-        <meta
-          property='og:image'
-          content='https://reactbkk.github.io/2.0.0/static/og-image.jpg'
-        />
-        <meta name='apple-mobile-web-app-title' content='ReactBKK' />
-        <link rel='shortcut icon' type='image/png' href='static/favicon.png' sizes='32x32' />
-        <link rel='apple-touch-icon' href='https://reactbkk.github.io/2.0.0/static/icon-ios-fullsize.png' />
-        <link rel='apple-touch-icon' sizes='152x152' href='https://reactbkk.github.io/2.0.0/static/icon-ios-152.png' />
-        <link rel='apple-touch-icon' sizes='167x167' href='https://reactbkk.github.io/2.0.0/static/icon-ios-167.png' />
-        <link rel='apple-touch-icon' sizes='180x180' href='https://reactbkk.github.io/2.0.0/static/icon-ios-180.png' />
-        <script dangerouslySetInnerHTML={{ __html: ga }} />
-        <link rel='manifest' href='static/manifest.json' />
-      </Head>
-      <style global jsx>{`
-        @font-face {
-          font-family: 'ThaiSans Neue';
-          src: local('ThaiSans Neue'),
-               url('static/thaisansneue/thaisansneue-bold-webfont.woff2') format('woff2'),
-               url('static/thaisansneue/thaisansneue-bold-webfont.woff') format('woff');
-          font-weight: bold;
-          font-style: normal;
-        }
-        @font-face {
-          font-family: 'ThaiSans Neue';
-          src: local('ThaiSans Neue'),
-               url('static/thaisansneue/thaisansneue-regular-webfont.woff2') format('woff2'),
-               url('static/thaisansneue/thaisansneue-regular-webfont.woff') format('woff');
-          font-weight: normal;
-          font-style: normal;
-        }
-        body {
-          margin: 0;
-          background: #222;
-          color: #fff;
-          font-family: ThaiSans Neue;
-          font-size: 22px;
-          -webkit-font-smoothing: antialiased;
+const checkDomain = (url) => url.replace(/([A-Za-z]):\/\//, '$1').split('/')[0]
+const isExternal = (url) => (url && url.indexOf(':') > -1 && checkDomain(window.location.href) !== checkDomain(url))
+const isKeyPressedWithClick = (event) => event.ctrlKey || event.altKey || event.shiftKey || event.metaKey
+
+export default class MainLayout extends React.Component {
+  componentDidMount () {
+    this.listenerClickLink()
+  }
+
+  listenerClickLink () {
+    [].forEach.call(document.querySelectorAll('a'), (el) => {
```

Comment:
There is a problem with this approach: Dynamically generated links will not be handled.

Instead, you can add a single `onClick` listener on the root `<div>` and let the `click` event on the link bubble up to that `<div>`. Then you can check if the thing being clicked is really a link.